src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -847,42 +848,134 @@ private void RebuildRegionConnections()
+            }
+        }
+    }
+
+    private Line2D[] BuildUnknownRegionConnections(Line2D startingConnection, PatchRegion targetRegion,
+        PatchRegion startRegion, Color color, bool reversed)
+    {
+        var startingPoint = reversed ?
+            startingConnection.Points[startingConnection.Points.Length - 1] :
+            startingConnection.Points[0];
+
+        var adjacencies = startRegion.PatchAdjacencies[targetRegion.ID];
+
+        // Generate a list of patches to connect to
+        var patches = targetRegion.Patches
+            .Where(p => p.Visibility == MapElementVisibility.Unknown)
+            .Where(p => adjacencies.Contains(p))
+            .ToArray();
+
+        var newConnections = new Line2D[patches.Length];
+
+        var i = 0;
+        foreach (var targetPatch in patches)
+        {
+            var halfNodeSize = Constants.PATCH_NODE_RECT_LENGTH / 2;
+            var endingPoint = targetPatch.ScreenCoordinates + Vector2.One * halfNodeSize;
+
+            // Draw a straight line if possible
+            if (endingPoint.x == startingPoint.x || endingPoint.y == startingPoint.y)
             {
-                ApplyFadeToLine(line, true);
+                var straightPoints = new[]
+                {
+                    startingPoint,
+                    endingPoint,
+                };
+
+                newConnections[i++] = CreateConnectionLine(straightPoints, color);
+                continue;
             }
+
+            var intermediate = new Vector2(endingPoint.x, startingPoint.y);
+
+            var points = new[]
+            {
+                startingPoint,
+                intermediate,
+                endingPoint,
+            };
+
+            newConnections[i++] = CreateConnectionLine(points, color);
         }
+
+        return newConnections;
     }
 
-    private Vector2 GetUnknownRegionPosition(PatchRegion region)
+    private void ApplyFadeIfNeeded(PatchRegion startingRegion, PatchRegion endingRegion,
+        Line2D line, bool reversed)
     {
-        return RegionCenter(region);
+        // Do not apply fade from hidden or unknown region
+        if (startingRegion.Visibility != MapElementVisibility.Shown)
+            return;
+
+        // Do not apply fade if target region is visible
+        if (endingRegion.Visibility == MapElementVisibility.Shown)
+            return;
+
+        // Apply fade if target region is hidden
+        if (endingRegion.Visibility == MapElementVisibility.Hidden)
+        {
+            ApplyFadeToLine(line, reversed);
+            return;
+        }
+
+        // Apply fade only if no connecting patches are visible in the target region
+
+        var adjacencies = startingRegion.PatchAdjacencies[endingRegion.ID];
+
+        var patches = endingRegion.Patches
+            .Where(p => p.Visibility == MapElementVisibility.Unknown)
+            .Where(p => adjacencies.Contains(p));
+
+        if (!patches.Any())
```

Comment:
This is kind of not fitting in with the overall thrive style where even these very long LINQ checks are just directly put in the if statement.

So this should be something like:
```
if (endingRegion.Patches
            .Where(p => p.Visibility == MapElementVisibility.Unknown)
            .Any(p => adjacencies.Contains(p)))
{
    ApplyFadeToLine(line, reversed);
}
```